src/Compilers/CSharp/Portable/Symbols/PublicModel/NamedTypeSymbol.cs

```diff
@@ -194,7 +194,11 @@ INamedTypeSymbol INamedTypeSymbol.TupleUnderlyingType
 
         bool INamedTypeSymbol.IsSerializable => UnderlyingNamedTypeSymbol.IsSerializable;
 
-        bool INamedTypeSymbol.IsFileLocal => UnderlyingNamedTypeSymbol.AssociatedSyntaxTree is not null;
+        bool INamedTypeSymbol.IsFileLocal =>
+            // Even though we round-trip the file identifier through metadata to support EE,
+            // we don't want public API to give the impression that the compiler considers a metadata type to be a file-local type.
+            UnderlyingNamedTypeSymbol.OriginalDefinition is not Metadata.PE.PENamedTypeSymbol
```

Comment:
>UnderlyingNamedTypeSymbol.OriginalDefinition is not Metadata.PE.PENamedTypeSymbol [](http://example.com/codeflow?start=12&length=81)

Perhaps we should check for what we expect instead of trying to filter out what we don't expect. If we are saying, only source symbols can be file local, then we should check if original definition is a source symbol.